README.md

```diff
@@ -70,24 +70,28 @@ When building packages that use CUDA, a tar package of TensorRT for the intended
+The `open-ce build env` command can then be used to build a collection of Open-CE packages. An Open-CE environment file needs to be passed in as input. A selection of environment files are provided within the [`open-ce-environments` repo](https://github.com/open-ce/open-ce-environments) for different frameworks such as TensorFlow and PyTorch. The output from running `open-ce build env` will be a local conda channel (by default called `condabuild`) and one or more conda environment file(s) in the output folder depending on the selected build configuration. For more details on `open-ce build env`, please see [`doc/README.open_ce_build.md`](doc/README.open_ce_build.md#open-ce-build-env-sub-command).
 
 The following commands will use the `opence-env.yaml` Open-CE environment file to build all of the Open-CE packages for Python 3.7 (the default), including CUDA builds and cpu-only builds (also the default). The commands should be run from within the same directory that contains `local_files`.
 
 ```bash
 # Clone Open-CE from GitHub
 git clone https://github.com/open-ce/open-ce.git
+# Clone Open-CE Environments from GitHub
+git clone https://github.com/open-ce/open-ce-environments.git
 # Build packages
-./open-ce/open-ce/open-ce build env open-ce/envs/opence-env.yaml
+./open-ce/open-ce/open-ce build env ./open-ce-environments/envs/opence-env.yaml
 ```
 
-The following commands will use opence-env.yaml Open-CE environment file from a specific Open-CE release to build all of the Open-CE packages for Python 3.6, 3.7 and 3.8, including only CUDA builds. The commands should be run from within the same directory that contains `local_files`.
+The following commands will use the `opence-env.yaml` Open-CE environment file from a specific Open-CE release to build all of the Open-CE packages for Python 3.6, 3.7 and 3.8, including only CUDA builds. The commands should be run from within the same directory that contains `local_files`.
 
 ```bash
 # Clone a specific Open-CE release from GitHub
 git clone https://github.com/open-ce/open-ce.git --branch open-ce-v1.0.0
+# Clone Open-CE Environments from GitHub
+git clone https://github.com/open-ce/open-ce-environments.git --branch open-ce-v2021.1.0
 # Build packages
-./open-ce/open-ce/open-ce build env --python_versions 3.6,3.7,3.8 --build_types cuda open-ce/envs/opence-env.yaml
+./open-ce/open-ce/open-ce build env --python_versions 3.6,3.7,3.8 --build_types cuda ./open-ce-environments/envs/opence-env.yaml
```

Comment:
Will these commands work with the `--docker_build` option. Will the environment files get passed into the container?

Reply:
No it won't. I forgot to look into that.
We may need to add another volume mount to the docker command.

Reply:
I got around it during some testing by cloning the environments repo into the open-ce folder.

Reply:
But with the absolute path conversion being handled earlier on, it won't work. I'll need to look into this more.

Reply:
Copying the environment files into the container is probably the best thing to do. That actually makes things a little more flexible than what we currently have as well. Probably need to copy in the environment files and the conda_build_config file.

Reply:
The only problem I see with this is that the env file imports other env files. Should we:

1.  fully resolve all imports and copy all those files in
2.  just copy all yaml files in the same folder as the env file
3.  copy all contents of the folder that contians the env file

Also, I don't recall. Can env files import files from anywhere in the file system?
I imagine if so, then dockerbuild doesn't work in general as it is now, so should we even worry about that?

Reply:
yeah, it looks like if it's given a full path anywhere, it'll try and load it, otherwise it looks in the standard location:

https://github.ibm.com/open-ce/open-ce/blob/master/open-ce/env_config.py#L91-L96

With this new model of the environments being separate, maybe we want to change it so it only looks in the specified open-ce environments repo. That would be more deterministic (and more secure). And I guess that would mean your option 3 (or I guess option 2 as well if we wanted to be more specific with it).

Reply:
I am now mounting the directory that contains the env files, and copying the conda build config into the container.

Regarding the env files importing from anywhere in the fs, this method has the same limitations the previous implementation had, so I think if we want to change this behavior, we change it in another PR.